nss/aad_auth/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/ubuntu/aad-auth/internal/logger"
+	"github.com/ubuntu/aad-auth/internal/nss"
+)
+
+func main() {
+	flag.Usage = aadAuthUsage
+	flag.Parse()
+
+	switch flag.Arg(0) {
+	case "getent":
+		ctx := nss.CtxWithSyslogLogger(context.Background())
+		defer logger.CloseLoggerFromContext(ctx)
+
+		db, key := flag.Arg(1), flag.Arg(2)
+
+		if !dbIsSupported(db) {
+			log.Fatalf("Request db %s is not supported", db)
+		}
+
+		out := Getent(ctx, db, key)
+		fmt.Print(out)
+	case "":
+		flag.Usage()
+		os.Exit(1)
+	default:
+		log.Fatalf("Invalid argument %s", flag.Arg(0))
```

Comment:
Same here, print the usage after the error message as the user doesn’t know how to use it :)

I suggest that you feed an err in both `case ""` and `default`, and then, out, (this will work for Getent too), you check if the err is not nil and print the error, then usage, then exit.